resources/waterheater.rb

```diff
@@ -790,14 +790,14 @@ def self.add_desuperheater(model, t_set, tank, desuperheater_clg_coil, wh_type,
       dsh_program.setName("#{tank_name} DSH Program")
       dsh_program.addLine("Set #{tank_name}_eta_c = #{eta_c}")
       dsh_program.addLine("Set Avail_Cap = #{reclaimed_efficiency} * (#{coil_clg_energy.name} + #{coil_elec_energy.name})")
-      dsh_program.addLine("If WarmupFlag") # need to initialize cummulative dsh energy number
-      dsh_program.addLine("Set #{dsh_total.name} = 0.0")
+      dsh_program.addLine("If WarmupFlag") # need to initialize cumulative dsh energy number
+      dsh_program.addLine("  Set #{dsh_total.name} = 0.0")
       dsh_program.addLine("Else")
-      dsh_program.addLine("Set #{dsh_total.name} = #{dsh_total.name} + Avail_Cap")
+      dsh_program.addLine("  Set #{dsh_total.name} = #{dsh_total.name} + Avail_Cap")
       dsh_program.addLine("EndIf")
       dsh_program.addLine("Set #{tank_name}_dsh_load_saving = -(@Min #{wh_energy.name} #{dsh_total.name})")
-      dsh_program.addLine("Set #{dsh_total.name} = #{dsh_total.name} + #{tank_name}_dsh_load_saving") # update cummulative dsh energy pool
-      dsh_program.addLine("Set #{dsh_actuator.name} = #{tank_name}_dsh_load_saving / (SystemTimeStep * 3600 ) / #{tank_name}_eta_c")
+      dsh_program.addLine("Set #{dsh_total.name} = #{dsh_total.name} + #{tank_name}_dsh_load_saving") # update cumulative dsh energy pool
+      dsh_program.addLine("Set #{dsh_actuator.name} = #{tank_name}_dsh_load_saving * #{ec_adj.round(5)} / (SystemTimeStep * 3600) / #{tank_name}_eta_c") # convert to water heater power savings
```

Comment:
Changes looks good to me. EC adjustment is now also applied to desuperheater energy saving.